### PR TITLE
chore(environments): Add `Project` model

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Step 1 of https://github.com/PostHog/meta/pull/176.

## Changes

This adds the `Project` model. It's not hooked up to anything yet, but we ensure every _new_ environment (i.e. old-style project) now gets a project.

## How did you test this code?

~~TODO: A few tests for `create_with_team`.~~ Added some tests for new behavior.

### CODE REVIEW

**Comment by Twixes** [comment]
Per chat with Tim over at [#team-infrastructure](https://posthog.slack.com/archives/C01MM7VT7MG/p1710174385293909), the migration _should_ be safe BUT it's good that the "Validate Django migrations" is so diligent – we're just going to bypass it in this particular case and force-merge.

File: posthog/models/project.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+if TYPE_CHECKING:
+    from .team import Team
+
+
+class ProjectManager(models.Manager):
+    def create_with_team(self, team_fields: Optional[dict] = None, **kwargs) -> Tuple["Project", "Team"]:
+        from .team import Team
+
+        with transaction.atomic():
+            common_id = Team.objects.increment_id_sequence()
```
Comment (benjackwhite) [inline posthog/models/project.py:15]: Nice

File: posthog/models/project.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            project = self.create(id=common_id, **kwargs)
+            team = Team.objects.create(
+                id=common_id, organization=project.organization, project=project, **(team_fields or {})
+            )
+            return project, team
+
+
+class Project(models.Model):
```
Comment (benjackwhite) [inline posthog/models/project.py:23]: Nit: Maybe we should add a comment here warning and explaining that this is part of a longer term plan? Thinking for newer people / people not super clued up here that they could accidentally start using it.

**Review by benjackwhite — APPROVED** [review]
Very clean implementation! Looks good to me

File: posthog/models/project.py (hunk 2 above)
Comment (Twixes) [reply posthog/models/project.py:23]: I do always appreciate a docstring. Added

**Pull request merged**